File: packages/backend/src/main/java/com/bookspank/backend/service/ClubHistoryService.java
package com.bookspank.backend.service;

import java.util.List;
import java.util.Optional;

import com.bookspank.backend.dto.CompleteBookRequest;
import com.bookspank.backend.model.ClubHistoryEntry;

public interface ClubHistoryService {
    public List<ClubHistoryEntry> getClubHistory(Integer limit, Integer offset);

    public Optional<ClubHistoryEntry> getCurrent();

    public Optional<ClubHistoryEntry> completeBook(CompleteBookRequest form);

}

File: packages/aws-nextjs/lib/service/client/ClubHistoryService.ts
import { ClubHistoryEntry } from "@/model/ClubHistoryEntry";

export interface IClubHistoryService {
    getClubHistory(limit: number, offset: number): Promise<ClubHistoryEntry[]>;
    getCurrent(): Promise<ClubHistoryEntry>;
    completeBook(bookId: number): Promise<ClubHistoryEntry>;
}